File: cdk-from-cfn-testing/expected/cloudwatch/csharp/CloudwatchConstruct.cs
using Amazon.CDK;
using Amazon.CDK.AWS.CloudWatch;
using Constructs;
using System.Collections.Generic;

namespace CloudwatchConstruct
{
    public class CloudwatchConstructProps
    {
        /// <summary>
        /// Environment used for this deployment.
        /// </summary>
        public string EnvironmentName { get; set; }

    }

    public class CloudwatchConstruct : Construct
    {
        public CloudwatchConstruct(Construct scope, string id, CloudwatchConstructProps props = null) : base(scope, id)
        {
            // Applying default props
            props ??= new CloudwatchConstructProps();
            props.EnvironmentName ??= "dev";


            // Resources
            var myApi5xxErrorsAlarm = new CfnAlarm(this, "MyApi5xxErrorsAlarm", new CfnAlarmProps
            {
                AlarmDescription = "Example alarm",
                Namespace = "AWS/ApiGateway",
                Dimensions = new []
                {
                    new CfnAlarm.DimensionProperty
                    {
                        Name = "ApiName",
                        Value = "MyApi",
                    },
                },
                MetricName = "5XXError",
                ComparisonOperator = "GreaterThanThreshold",
                Statistic = "Average",
                Threshold = 0.005,
                Period = 900,
                EvaluationPeriods = 1,
                TreatMissingData = "notBreaching",
                AlarmActions = new []
                {
                    Fn.ImportValue($"{props.EnvironmentName}AlarmsTopicArn"),
                },
            });
        }
    }
}

File: cdk-from-cfn-testing/expected/cloudwatch/typescript/CloudwatchConstruct.ts
import * as cdk from 'aws-cdk-lib';
import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';

export interface CloudwatchConstructProps {
  /**
   * Environment used for this deployment.
   * @default 'dev'
   */
  readonly environmentName?: string;
}

export class CloudwatchConstruct extends Construct {
  public constructor(scope: Construct, id: string, props: CloudwatchConstructProps = {}) {
    super(scope, id);

    // Applying default props
    props = {
      ...props,
      environmentName: props.environmentName ?? 'dev',
    };

    // Resources
    const myApi5xxErrorsAlarm = new cloudwatch.CfnAlarm(this, 'MyApi5xxErrorsAlarm', {
      alarmDescription: 'Example alarm',
      namespace: 'AWS/ApiGateway',
      dimensions: [
        {
          name: 'ApiName',
          value: 'MyApi',
        },
      ],
      metricName: '5XXError',
      comparisonOperator: 'GreaterThanThreshold',
      statistic: 'Average',
      threshold: 0.005,
      period: 900,
      evaluationPeriods: 1,
      treatMissingData: 'notBreaching',
      alarmActions: [
        cdk.Fn.importValue(`${props.environmentName!}AlarmsTopicArn`),
      ],
    });
  }
}
